runtime/lua/vim/lsp/completion.lua

```diff
@@ -706,6 +853,36 @@ local function enable_completions(client_id, bufnr, opts)
         end
       end,
     })
+
+    if vim.o.completeopt:find('popup') then
+      api.nvim_create_autocmd('Completechanged', {
+        group = group,
+        buffer = bufnr,
+        callback = function(args)
+          local completed_item = api.nvim_get_vvar('event').completed_item or {}
+          if (completed_item.info or '') ~= '' then
+            local data = vim.fn.complete_info({ 'selected' })
+            update_popup_window(data.preview_winid, data.preview_bufnr)
+            return
+          end
+
+          if
+            not lsp.get_client_by_id(
+              vim.tbl_get(completed_item, 'user_data', 'nvim', 'lsp', 'client_id')
+            )
+          then
+            return
+          end
+
+          local param = vim.tbl_get(completed_item, 'user_data', 'nvim', 'lsp', 'completion_item')
+          if param then
+            debounce_info_request(args.buf, param, completed_item.word)
```

Comment:
can we check `client:supports_method(ms.completionItem_resolve) `before sending the request? I'm seeing many `vim.notify` warnings for some LSP servers (e.g. json language service) which don't support `completionItem/resolve`: they simply provide documentation in the original response.